kernel/share.c

```diff
@@ -587,6 +595,21 @@ static int lock_unlock
 	}
 }
 
+static uint16_t is_file_open(char FAR * filename)
+{
+	int i;
+	file_t FAR *fptr;
+
+	for (i = 0; i < file_table_size; i++) {
+		fptr = &file_table[i];
+		if (fptr->filename[0] == '\0')
+			continue;
+		if (fstrncmp(filename, fptr->filename, sizeof(*fptr->filename)) == 0)
```

Comment:
Why asterisk at sizeof?

Reply:
I thought `sizeof(fptr->filename)` would give me the size of the pointer, but I now think I'm wrong.

Reply:
But with asterisk it would give
sizeof(fptr->filename[0]), which is even worse.
I think sizeof() on far arrays does work in
fdpp right now (but can't be 100% sure
as I reworked it many times, and don't
remember its current state).

Reply:
I repushed with `fnmatches()` with which both the pointer and the sizeof are no longer necessary.